docs/classification/micro_macro.md

```diff
@@ -0,0 +1,201 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.1
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Micro and Macro Averaging
+
+`Precision` and `Recall` are two commonly used metrics to assess the performance of a classification model. The metrics are fairly intuitive with binary classification. But when it comes to multi-class classification these metrics need to be tweaked a bit to measure performance of each class. 
+
+This tutorial explains the concepts of `micro` and `macro` averaging and the scenarios in which to use each of the two approaches.
+
++++
+
+## Basic Concepts
+
+Precision is the proportion of positive identifications that were actually correct.
+
+Precision = $\frac{True   Positives}{True   Positives + False   Positives}$
+
+Recall is the proportion of the actual positives that were correctly identified.
+
+Recall = $\frac{True   Positives}{True   Positives + False   Negatives}$
+
++++
+
+Now let's load some data and train a classifier.
+
+```{code-cell} ipython3
+from sklearn.datasets import load_iris as load_data, make_classification
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+from sklearn_evaluation import plot
+
+X, y = load_data(return_X_y=True)
+
+# split data into train and test
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=0)
+
+y_pred = RandomForestClassifier(random_state=0).fit(X_train, y_train).predict(X_test)
+
+target_names = ["A", "B", "C"]
+```
+
+## Micro-average approach
+
+The `micro-average` precision and recall score is calculated from the individual classes’ `true positives (TPs)`, `true negatives (TNs)`, `false positives (FPs)`, and `false negatives (FNs)` of the model. It aggregates the contributions of all classes to compute the average metric and weights each sample equally.
+
+Let's plot the confusion matrix and derive each metric.
+
+```{code-cell} ipython3
+plot.ConfusionMatrix.from_raw_data(y_test, y_pred, target_names=target_names)
+```
+
+Metrics for micro-average precision:
```

Comment:
add a note here so it's easier to connect TP/TN to their position in the confusion matrix. i.e., TP are in the main diagonal, TN is the sum of values on each  column (excluding the one in the main diagonal)